cmd/controller-manager/app/controllermanager.go

```diff
@@ -357,3 +452,20 @@ func setupClusterAPIClusterDetector(mgr controllerruntime.Manager, opts *options
 
 	klog.Infof("Success to setup cluster-api cluster detector")
 }
+
+// IsControllerEnabled check if a specified controller enabled or not.
+func (c ControllerContext) IsControllerEnabled(name string) bool {
```

Comment:
Now you can delete the `IsControllerEnabled` function in `options.go`

Reply:
> cc @iawia002 @lonelyCZ
> 
> @jameszhangyukun The enabled controller list is still `hard-coded`, are you planning to do it by following PR? I'm ok to integrate it by following PR, just a reminder that issue #1093 will be closed by this PR since the `Fixes #1085` is present in the PR description.

I try to do it in this pr. I will fix it ASAP

Reply:
+1